sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -81,12 +81,16 @@ object DataSourceUtils extends PredicateHelper {
    * in a driver side.
    */
   def verifySchema(format: FileFormat, schema: StructType): Unit = {
+    checkFieldType(format, schema)
```

Comment:
Doesn't this affect other sources too? not only parquet? After this PR, it won't validate the field names against https://github.com/apache/spark/blob/62cf4d41b12a3a6d94d011a5b76e66ccaa3fed2a/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala#L171 anymore. It change the semantic of an API